weaviate/collections/data.py

```diff
@@ -220,14 +220,9 @@ def _reference_add_many(self, refs: List[DataReferences]) -> BatchReferenceRetur
         ]
         return self._batch_rest.references(list(batch))
 
-    def _reference_delete(
-        self, from_uuid: UUID, from_property: str, ref: WeaviateReference
-    ) -> None:
+    def _reference_delete(self, from_uuid: UUID, from_property: str, ref: _Reference) -> None:
```

Comment:
why are these types _Reference again? They should be Union[_Reference, UUID, Sequence[UUID]]

Reply:
this was on purpose to allow giving UUIDS/str directly without a unneeded extra class

Reply:
(same for all of the other functions)